src/v/cloud_roles/signature.h

```diff
@@ -58,17 +57,23 @@ class time_source {
     ~time_source() noexcept = default;
 
     /// Return formatted date in ISO8601 format
+    /// Example: 20201231
     ss::sstring format_date() const;
 
     /// Return formatted date in ISO8601 format
+    /// Example: 20201231T123100Z
     ss::sstring format_datetime() const;
 
+    /// Return formatted datetime according to RFC9110
+    /// Example: Tue, 15 Nov 2010 08:12:31 GMT
+    ss::sstring format_http_datetime() const;
+
 private:
     template<class Fn>
     explicit time_source(Fn&& fn, int);
 
     /// Format date-time according to format string
-    ss::sstring format(const char* fmt) const;
+    ss::sstring format(auto fmt) const;
```

Comment:
what does it mean to have auto parameter. does that mean format is now implicitly a template? if so, does it need to be?